src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -425,32 +424,46 @@ string GetHeaderSeparator (string name)
 			return value;
 		}
 
+		void AddManagedHeaders (NSMutableDictionary nativeHeaders, IEnumerable<KeyValuePair<string, IEnumerable<string>>> managedHeaders)
+		{
+			foreach (var keyValuePair in managedHeaders) {
+				var keyPtr = NSString.CreateNative (keyValuePair.Key);
+				var valuePtr = NSString.CreateNative (string.Join (GetHeaderSeparator (keyValuePair.Key), keyValuePair.Value));
```

Comment:
This is more of a curiosity than a statement, but would using an `NSMutableString` and appending chunks to it be faster? Could be benchmarked, I think.